_posts/2025-10-14-cqrs.adoc

```diff
@@ -0,0 +1,374 @@
+We can e.g. embed Debezium into our custom microservice, but in such case we have to implement sink part (part which writes the captured data into target database) ourselves.
+
+The following diagram illustrates this setup:
+
+++++
+<div class="imageblock centered-image">
+	<img src="/assets/images/2025-10-14-cqrs/cqrs-debezium.png" class="responsive-image" alt="Demo application with Debezium-based replication">
+</div>
+++++
+
+For simplicity, in out demo application we will also include Kafka Connect as the Debezium runtime.
+We will show two examples of using Debezium for data replication:
+
+* Replicating data from PostgreSQL to MySQL using the Debezium JDBC Sink Connector.
+* Replicating data from PostgreSQL to QuestDB using the QuestDB Kafka Sink Connector.
+
+=== Replicating from PostgreSQL to MySQL
+
+This setup demonstrates how to use Debezium with Kafka Connect to replicate data between two relational databases and target database supports JDBC API.
+
+The source side uses the Debezium PostgreSQL connector with a minimal configuration that specifies connection details and the tables to monitor:
+
+[source,json,options="nowrap"]
+----
+{
+    "name": "postgres-source",
+    "config": {
+        "connector.class": "io.debezium.connector.postgresql.PostgresConnector",
+        "tasks.max": "1",
+        "database.hostname": "postgres",
+        "database.port": "5432",
+        "database.user": "postgres",
+        "database.password": "postgres",
+        "database.dbname": "postgres",
+        "topic.prefix": "polls",
+        "table.include.list": "public.optionentity, public.pollentity, public.pollentity_optionentity, public.pollentity_voteentity, public.voteentity"
+    }
+}
+----
+
+This configuration captures all changes in the specified tables and publishes them to Kafka topics under the prefix `polls`.
+
+On the sink side, we use the Debezium JDBC Sink Connector to insert these records into a MySQL database.
+Since the JDBC sink understands Debezium’s event format, no single message transformation (SMT) is required.
+
+[source,json,options="nowrap"]
+----
+{
+    "name": "mysql-sink",
+    "config": {
+        "connector.class": "io.debezium.connector.jdbc.JdbcSinkConnector",
+        "tasks.max": "1",
+        "connection.url": "jdbc:mysql://mysql:3306/polls",
+        "connection.username": "mysqluser",
+        "connection.password": "mysqlpw",
+        "insert.mode": "upsert",
+        "delete.enabled": "true",
+        "primary.key.mode": "record_key",
+        "schema.evolution": "basic",
+        "collection.name.format": "${source.table}",
+        "topics": "polls.public.voteentity, polls.public.optionentity, polls.public.pollentity, polls.public.pollentity_optionentity"
+    }
+}
+----
+
+In this configuration, the `insert.mode` is set to `upsert`, which ensures that both new inserts and updates are properly reflected in the target database.
+Deletes are enabled as well, since Hibernate may drop tables.
```

Comment:
I belive dropping tables is unrelated to this setting. Also drop table does not emit deletes for recors so the reference is probably not correct.